Allow for multiple staging environments

ttung opened the pull request:
### Description
Creates an easy-to-setup staging environment.  This allows developers to create arbitrary staging environments without setting up a new secret.  The callback URL is determined by extracting the Elastic Beanstalk environment name.  Note that this is actually dependent on the cname, but if all the instructions are followed, the cname should match the environment name.

Also updated docs to reflect the new setup.

### Test plan
1. Created and deployed to an AWS Elastic Beanstalk environment.
2. Run locally.

phoenixAja commented on src/py/aspen/aws/session.py:
```diff
@@ -0,0 +1,16 @@
+_session: Optional[Session] = None
+
+
+def session() -> Session:
```
since we have many different sessions (one for db, one for app, and now one for aws) i wonder if we should rename to something more specific? Like `aws_session`? Interested to hear what you think

ttung replied:
I used it as `aws.session()` in all the places to make it explicit.   Google's python style guide has some [recommendations](https://google.github.io/styleguide/pyguide.html#224-decision), but I don't know how to enforce that.  We'll just have to do it by convention...

phoenixAja commented on src/py/aspen/config/config.py:
```diff
@@ -29,7 +30,16 @@ def DEBUG(self):
     @property
     def SECRET_KEY(self):
-        return os.environ.get("SECRET_KEY")
+        # TODO: This probably needs to be sourced from secrets.  Somehow.  Who knows.
```
yeah that's a good idea to store in secrets, but then we would need an auth0 secret and an app secret, unless we create a mega secret. this definitely should be more secure for a production deploy

ttung replied:
yeah, let me fix this.  I don't think we should be committing this as is.

phoenixAja replied:
i think we can keep the env variable option for now, or maybe have a randomly generated string?

ttung replied:
ok i'm setting it to a random value for now.  I'll clean this up in a subsequent PR.

phoenixAja commented on src/py/aspen/config/config.py:
```diff
@@ -29,7 +30,16 @@ def DEBUG(self):
+        # TODO: This probably needs to be sourced from secrets.  Somehow.  Who knows.
+        return "abcabcabc"
+
+    @property
+    def SESSION_TYPE(self):
+        # TODO: This needs to be updated to something more sane to persist sessions
+        # across restarts.
+        # https://flask-session.readthedocs.io/en/latest/#sqlalchemysessioninterface is
+        # probably a sane option.
+        return "filesystem"
```
this is interesting. Do we need to persist sessions across restarts? I think if something like that happens sqlalchemy would automatically role back the session right?

ttung replied:
No, it wouldn't.  It'd force people to hit auth0 again, I believe.

ttung replied:
Ok so it looks like we don't need this.  From my testing, having a random value for `SECRET_KEY` is not great because it forces people to return to auth0, so we will want some persistent value for that.

Commit:
Allow for multiple staging environments

### Description
Creates an easy-to-setup staging environment.  This allows developers to create arbitrary staging environments without setting up a new secret.  The callback URL is determined by extracting the Elastic Beanstalk environment name.  Note that this is actually dependent on the cname, but if all the instructions are followed, the cname should match the environment name.

Also updated docs to reflect the new setup.

### Test plan
1. Created and deployed to an AWS Elastic Beanstalk environment.
2. Run locally.
